sqlgen/db.go

```diff
@@ -252,6 +292,36 @@ func (db *DB) BaseQuery(ctx context.Context, query *BaseSelectQuery) ([]interfac
 
 	clause, args := selectQuery.ToSQL()
 
+	if db.panicOnNoIndex && (query.Options == nil || !query.Options.AllowNoIndex) {
+		// We run an explain first and panic if there's no index
+		explained := "EXPLAIN " + clause
+		res, err := db.QueryExecer(ctx).QueryContext(ctx, explained, args...)
+		if err != nil {
+			return nil, err
+		}
+
+		explainRes, err := parseExplainResults(res)
+		res.Close()
+
+		// If our explain parsing fails, we don't analyze the response or fail the query. We know
+		// this works for our happy paths and nested queries are currently out of scope.
```

Comment:
Could we ever support nested querries?

Reply:
@er9781 I don't think nested queries are an issue for parsing, actually - this turned out to be a red herring when we were testing.

Reply:
Yeah, we  should get multiple rows in explain and we check all of them 👍